pkg/ccl/backupccl/restore_data_processor.go

```diff
@@ -502,6 +509,12 @@ func (rd *restoreDataProcessor) runRestoreWorkers(
 					return done, nil
 				}
 
+				ctx := logtags.AddTag(ctx, "restore-span", entry.ProgressIdx)
+				ctx, undo := pprofutil.SetProfilerLabelsFromCtxTags(ctx)
+				defer undo()
+				ctx, sp := tracing.ChildSpan(ctx, "restore.processRestoreSpanEntry")
+				defer sp.Finish()
```

Comment:
This change makes it better for backporting, but I'll admit to having to do a double-take during review about having defers in a for loop because the diff context didn't include the `done, err := func() ...`.